chart/templates/deployment.yaml

```diff
@@ -32,6 +32,14 @@ spec:
 {{ toYaml .Values.imagePullSecrets | indent 6 }}
 {{- end }}
       affinity:
+        {{- if .Values.nodeAffinity }}
+        nodeAffinity:
+          {{- toYaml . | nindent 10 }}
+        {{- else }}
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms: {{ include "linux-node-selector-terms" . | nindent 14 }}
```

Comment:
Hmm, I don't like the current behavior of this PR that drops the Linux-only tolerations + nodeSelectorTerms entirely if the user attempts to supply them.

I think that we should use the Helm [`mustMerge`](https://github.com/Masterminds/sprig/blob/master/docs/dicts.md#merge-mustmerge) functionality to take the output of the helpers.tpl function and merge them with whatever the user provides.

What do you think?

Reply:
I think so, I'm fixing it.

Reply:
@aiyengar2 
I have an update:

I've tried to merge the tolerations field with "linux-node-tolerations" variable in case of users supply the config, but for nodeAffinity, I've tried with the merge function, but it does not work as expected. Could you help to suggest about this?

Reply:
CC: @dramich